### PR TITLE
update(falco): add deprecation notice for -T, -t and -D

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind documentation

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area engine

**What this PR does / why we need it**:

As per discussion in https://github.com/falcosecurity/falco/issues/3174#issuecomment-2078841689 . This only deprecates the CLI option and offers an alternative, does not remove it.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update(falco): deprecate options -T, -t and -D
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Shouldn't we also warn the user through a log when these cmdline options are used?

**Comment by LucaGuerra** [comment]
@FedeDP sure, added a commit for it :)

File: userspace/falco/app/options.cpp [hunk 1]
```diff
@@ -106,12 +107,18 @@ bool options::parse(int argc, char **argv, std::string &errstr)
...
+		falco_logger::log(falco_logger::level::WARNING, "The -t option is deprecated and will be removed in Falco 0.39.0. Use -o rules[].disable.rule=* -o rules[].enable.tag=<tag> instead.");
 		for(auto &tag : m_cmdline_parsed["t"].as<std::vector<std::string>>())
 		{
 			enabled_rule_tags.insert(tag);
 		}
 	}
 
+	if(disabled_rule_substrings.size() > 0)
+	{
+		falco_logger::log(falco_logger::level::WARNING, "The -d option is deprecated and will be removed in Falco 0.39.0. Use -o rules[].disable.rule=<wildcard-pattern> instead.");
```
Comment (incertum) [inline userspace/falco/app/options.cpp:119]:
```suggestion
		falco_logger::log(falco_logger::level::WARNING, "The -D option is deprecated and will be removed in Falco 0.39.0. Use -o rules[].disable.rule=<wildcard-pattern> instead.");
```

The `-d` was deprecated a while ago and meant daemon mode.

Comment (LucaGuerra) [reply userspace/falco/app/options.cpp:119]: If I had a cent for every time I mixed `-d`, `-t`, `-T`, `-D`... :D good catch!

Comment (incertum) [reply userspace/falco/app/options.cpp:119]: Good thing that we deprecate them ...

**Review by incertum — APPROVED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**